fiftyone/core/view.py

```diff
@@ -278,6 +278,7 @@ def _aggregate(
 
         _frames_pipeline = []
         for s in self._stages:
+            s.validate(self)
```

Comment:
@benjaminpkane I think this may have been the reason that the unit test bug regarding excluding fields was observed.

Why is this line of code necessary? If a `ViewStage` has already been added to a view, then we know that `validate()` has already been called.

The reason I added `validate()` is so that view stages can do some "build time" validation that they are applicable to a given `SampleCollection` (eg that the appropriate field exists). This validation is always called when `add_stage()` is invoked, which should 

This did mean that I had to change the way some views were constructed in `core/session.py` and `core/main.py`, which sometimes created views by directly modifying `view._stages`, which is bad because it skips validation.

As I'm typing this, it occurs to me that calling `view.add_stage()` might not work for some use cases b/c it creates a new view object...

Anyway, we need to come to a consensus about when `validate()` is called, specifically whether it is done every time a view is aggregated or only once when the stage is added to the view.

Reply:
I just left a todo about this